libr/bin/format/mdmp/mdmp.c

```diff
@@ -1,371 +1,296 @@
+  obj->hdr = (struct minidump_header *)obj->b->buf;
 
-int r_bin_mdmp_init(struct r_bin_mdmp_obj *obj) {
-	if (obj->b->length < sizeof(MINIDUMP_HEADER)) {
-		eprintf("Error in r_bin_mdmp_init: length too short, not enough space for MINIDUMP_HEADER\n");
-		return false;
-	}
+  sdb_num_set(obj->kv, "mdmp.hdr.time_date_stamp", obj->hdr->time_date_stamp, 0);
+  sdb_num_set(obj->kv, "mdmp.hdr.flags", obj->hdr->flags, 0);
 
-	obj->hdr = (PMINIDUMP_HEADER)obj->b->buf;
+  if (obj->hdr->number_of_streams == 0) {
+    eprintf("Warning: No streams present!\n");
+    return false;
+  }
 
-	if (obj->hdr->NumberOfStreams <= 0) {
-		eprintf("Error in r_bin_mdmp_init: no streams\n");
-		return false;
-	}
+  if (obj->hdr->stream_directory_rva < sizeof(struct minidump_header))
+  {
+    eprintf("Error: RVA for directory resides in the header!\n");
+    return false;
+  }
 
-	if (obj->hdr->StreamDirectoryRva < sizeof(obj->hdr)) {
-		eprintf("Error in r_bin_mdmp_init: invalid StreamDirectoryRva, size %d\n", obj->hdr->StreamDirectoryRva);
-		return false;
-	}
+  if (obj->hdr->check_sum != 0) {
+    eprintf("TODO: Checksum present but needs validating!\n");
+    return false;
+  }
 
-	if (obj->hdr->CheckSum != 0) {
-		eprintf("Warning in r_bin_mdmp_init: CheckSum present, but not validated, because validation not implemented yet\n");
-	}
+  return true;
+}
 
-	sdb_num_set (obj->kv, "mdmp.hdr.TimeDateStamp", obj->hdr->TimeDateStamp, 0);
-	sdb_num_set (obj->kv, "mdmp.hdr.Flags", obj->hdr->Flags, 0);
 
-	if (!r_bin_mdmp_init_streams(obj)) {
-		return false;
-	}
-	return true;
+static bool r_bin_mdmp_init_directory_entry(struct r_bin_mdmp_obj *obj, struct minidump_directory *entry) {
+  int i;
+
+  struct minidump_handle_operation_list *handle_operation_list;
+  struct minidump_memory_list *memory_list;
+  struct minidump_memory64_list *memory64_list;
+  struct minidump_memory_info_list *memory_info_list;
+  struct minidump_module_list *module_list;
+  struct minidump_thread_list *thread_list;
+  struct minidump_thread_ex_list *thread_ex_list;
+  struct minidump_thread_info_list *thread_info_list;
+  struct minidump_unloaded_module_list *unloaded_module_list;
+
+  struct avrf_handle_operation *handle_operations;
+  struct minidump_memory_descriptor *memories;
+  struct minidump_memory_descriptor64 *memories64;
+  struct minidump_memory_info *memory_infos;
+  struct minidump_module *modules;
+  struct minidump_thread *threads;
+  struct minidump_thread_ex *ex_threads;
+  struct minidump_thread_info *thread_infos;
+  struct minidump_unloaded_module *unloaded_modules;
+
+  /* We could confirm data sizes but a malcious MDMP will always get around
+   * this! But we can ensure that the data is not outside of the file */
+  if (entry->location.rva + entry->location.data_size > obj->b->length)
+  {
+    eprintf("ERROR: Size Mismatch - Stream data is larger than file size!\n");
+    return false;
+  }
+
+  switch (entry->stream_type) {
+    case THREAD_LIST_STREAM:
```

Comment:
cases must be indented at the same level as switch, see r2 indentation style guidelines